docs/blueprints/forgotten-deletes.md

```diff
@@ -0,0 +1,124 @@
+# Forgotten Deletes
+
+## Abstract
+
+In the event of a node being audited for a piece which the satellite previously asked it to delete via delete pieces request, the node will present a signed message proving the legitimacy of the deletion. The node is not penalized and the object is deleted. We eliminate the potential of forgotten garabage collection deletes entirely.
+
+## Background
+
+In the event of a DB restoration from backup, it may be possible for the satellite to restore pointers for objects which were deleted, or revert pointers to a previous state containing nodes which no longer have those pieces. In this situation, the satellite would audit nodes for pieces which they were told to delete which may result in unfair disqualifications.
+
+## Design
+
+The solution to this problem is twofold, as there are two ways in which pieces are deleted: delete pieces requests and garbage collection.
+
+### Garbage collection
+
+Garbage collection works by observing the metainfo loop. This collects all the piece IDs which a node should have according to the pointer DB at that time. The satellite then sends these piece IDs to the storage node, indicating that it should keep these pieces and all pieces which were created after the creation of the bloom filter and delete the rest.
+
+Instead of proving the pieces were deleted, we will remove the possibility of these pieces being restored.
+To acheive this we will simply run garbage collection from a DB snapshot at a point in time beyond which we will never restore. That is, we must decide what is the furthest possible point the past we may restore to and only run garbage collection on a snapshot at or before this point. If this is guaranteed, no pieces deleted by garbage collection should ever be reverted by a DB restoration.
+
+### Delete pieces requests
+
+Delete pieces requests are issued when a user deletes an object. The metainfo service deletes the object's pointers and then sends delete pieces requests to each of the reliable nodes which held pieces. The rest of the nodes' pieces will be eventually deleted by garbage collection. 
+
+To prove to the satellite that a piece was deleted legitimately, we will have it sign the delete pieces requests. The node will present these to the satellite in the event of an audit for a deleted piece.
+
+```
+message DeletePiecesRequest {
+    repeated bytes piece_ids = 1 [(gogoproto.customtype) = "PieceID", (gogoproto.nullable) = false];
+    bytes satellite_signature = 2;
+}
+```
+
+The storage nodes will serialize and save these signed messages into a file store.
+
+#### File store
+
+// TODO: directory names
+Within the storage node's 'storage' directory, we will add a new directory called 'deleted-pieces-proof'. Within this directory we will create a subdirectory for each satellite, and within each of these we will add two more directories: 'pieces' and 'signed-requests'.
```

Comment:
nit - backticks instead of single quotes around `storage`, `deleted-pieces-proof`, `pieces`, `signed-requests`